wallet/wallet.go

```diff
@@ -236,13 +236,46 @@ func (w *Wallet) readDBVoteBits(dbtx walletdb.ReadTx) stake.VoteBits {
 	return vb
 }
 
+func (w *Wallet) readDBTicketVoteBits(dbtx walletdb.ReadTx, ticketHash *chainhash.Hash) *stake.VoteBits {
+	var bits uint16 = 0x0001
+	var hasSavedPrefs bool
+
+	version, deployments := CurrentAgendas(w.chainParams)
+	for i := range deployments {
+		d := &deployments[i]
+		choiceID := udb.TicketAgendaPreference(dbtx, ticketHash, version, d.Vote.Id)
+		if choiceID == "" {
+			continue
+		}
+		hasSavedPrefs = true
+		for j := range d.Vote.Choices {
+			choice := &d.Vote.Choices[j]
+			if choiceID == choice.Id {
+				bits |= choice.Bits
+				break
+			}
+		}
+	}
+
+	if !hasSavedPrefs {
+		return nil
+	}
+
+	tvb := stake.VoteBits{
+		Bits:         bits,
+		ExtendedBits: make([]byte, 4),
+	}
+	binary.LittleEndian.PutUint32(tvb.ExtendedBits, version)
+	return &tvb
```

Comment:
Looks like the only caller of this will just immediately dereference this pointer (when it's not nil), we should probably return the non-pointer here and add another `ok bool` return.